frontend/src/middlewares/error-handling/error-handling-middleware.ts

```diff
@@ -0,0 +1,22 @@
+import { type Middleware, type PayloadAction } from '@reduxjs/toolkit';
+import { toast } from 'react-toastify';
+
+type RejectedAction = PayloadAction<
+  unknown,
+  string,
+  unknown,
+  Error | undefined
+>;
+
+const handleAction =
+  (next: (action: RejectedAction) => RejectedAction) =>
+  (action: RejectedAction): RejectedAction => {
+    if (action.type.endsWith('/rejected') && action.error) {
+      toast.error(action.error.message);
```

Comment:
Let's create a service for notifications. You can check how it was implemented in the mini project